packages/components/src/components/languageSupport/highlightingStyle.ts

```diff
@@ -0,0 +1,113 @@
+import { HighlightStyle } from "@codemirror/language";
+import { tags } from "@lezer/highlight";
+
+// TODO: Freeze color theme after experimentation
+
+const clampComponent = (a: number) => (a < 255 ? (a >= 0 ? a : 0) : 255);
+const saturation = 0.7;
+const lightness = -10;
+
+const colors = [
+  "#bea066", // numbers
+  "#ba5a61", // names
+  "#8797a1", // operators
+  "#d4d4d4", // invalid
+  "#8e949f", // separators
+  "#68707f", // comments
+  "#5091c7", // variables
+  "#247ea2", // strings
+  "#ae8055", // constants
+  "#a564b8", // keywords
+].map(
+  (a) =>
+    "#" +
+    [
+      [1, 3],
```

Comment:
This is quite magical. It think I get what you are doing (modifying lightness and saturation of the constant colours?) But do you mind if you could either:
1. (preferably) hardcode the final colours calculated or
2. Extract this into a named utility function? The amount of magic numbers makes this annoying to read.

Reply:
> (preferably) hardcode the final colours calculated or

Yeah, that's what I am going for — line 4 here reminds me to do that afterwards)